dropbear: more fixes

rockdrilla opened the pull request:
Changes:
- fix missing depends for sntrup761x25519-sha512
- don't limit channel window to 500MB
- disable login recording with musl libc

These changes should completely resolve https://github.com/openwrt/openwrt/issues/19566.

rockdrilla commented:
@hauke, @oldium, @grosjo PTAL.

map-b commented:
Related #15417

oldium commented:
> Related #15417

Yes, the patch there fixes the paths in Musl libc library, but it does not implement the missing utmp/wtmp library functionality. Without that it does not make much sense to enable it.

@rockdrilla I think the PKG_RELEASE needs an update

rockdrilla commented:
@oldium I've already increased PKG_RELEASE.

grosjo commented:
Would be nice to reconsider https://github.com/openwrt/openwrt/pull/15417 and merge it alltogether

rockdrilla commented:
I dont't see much effort on reimplementing all that code (it's more than just "right paths").

oldium commented:
It is not about the paths. The library functions are empty - this is why the paths are invalid - to prevent applications from using them. You cannot fix one part of the problem and ignore that the function implementations are empty. Doing so will lead to reports from users complaining that the login tracking does not work for them.

You are of course free to use the `/var/log/utmp` and similar paths variables in your own programs, but on Musl you are left completely on your own (i.e. you need to re-implement it or use a different approach). Dropbear _is_ using library functions (like `setutent` and `endutent` for example), so “fixing” the paths does not make sense even from the Dropbear point of view.

The easiest “fix” is to use different libc implementation, like glibc.

rockdrilla commented:
It seems that we need that too: https://github.com/mkj/dropbear/pull/377

grosjo commented:
Will this ever be pushed to the tree ?

rockdrilla committed:
dropbear: fix missing depends for sntrup761x25519-sha512

resolves FTBFS where internal option DROPBEAR_SHA512 wasn't enabled by any of options

Signed-off-by: Konstantin Demin <rockdrilla@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19715
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

rockdrilla committed:
dropbear: disable login recording with musl libc

disallow lastlog/login()/utmp/wtmp with musl libc in order to avoid runtime errors due to musl implementation details.

ref: https://github.com/openwrt/openwrt/issues/19566#issuecomment-3167897095

Signed-off-by: Konstantin Demin <rockdrilla@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19715
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

rockdrilla committed:
dropbear: don't limit channel window to 500MB

cherry-pick commit from https://github.com/mkj/dropbear/pull/377

Signed-off-by: Konstantin Demin <rockdrilla@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/19715
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

rockdrilla commented:
Ready for review.

Changes from past version: I've removed "extra" checks in `053-Don-t-limit-channel-window-to-500MB.patch` - not needed and even may be erroneous.

grosjo commented:
WHo can do the review ? @nbd168
